src/Microsoft.Dnx.Watcher.Core/ProcessWatcher.cs

```diff
@@ -0,0 +1,107 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Dnx.Watcher.Core
+{
+    public class ProcessWatcher
+    {
+        private Process _runningProcess;
+
+        public int Start(string executable, string arguments, string workingDir)
+        {
+            // This is not thread safe but it will not run in a multithreaded environment so don't worry
+            if (_runningProcess != null)
+            {
+                throw new InvalidOperationException("The previous process is still running");
+            }
+
+            // TODO: check that it is running 
+            _runningProcess = new Process();
+            _runningProcess.StartInfo = new ProcessStartInfo()
+            {
+                FileName = executable,
+                Arguments = arguments,
+                UseShellExecute = false,
+                WorkingDirectory = workingDir
+            };
+
+            WorkaroundDnxInternalEnvironmentVariables(_runningProcess.StartInfo);
+
+            _runningProcess.Start();
+
+            return _runningProcess.Id;
+        }
+
+        public async Task<int> WaitForExitAsync(CancellationToken cancellationToken)
+        {
+            try
+            {
+                await Task.Run(() =>
+                {
+                    while (true)
+                    {
+                        if (_runningProcess.WaitForExit(500))
+                        {
+                            break;
+                        }
+
+                        if (cancellationToken.IsCancellationRequested)
+                        {
+                            if (!_runningProcess.HasExited)
+                            {
+                                _runningProcess.Kill();
+                            }
+
+                            break;
+                        }
+                    }
+                });
+
+                return _runningProcess.ExitCode;
+            }
+            finally
+            {
+                _runningProcess = null;
+            }
+        }
+
+        // This must be removed while DNX doesn't pass environment variables anymore
+        private static void WorkaroundDnxInternalEnvironmentVariables(ProcessStartInfo procStartInfo)
+        {
+            string[] _environmentVariablesToRemove = new string[]
+            {
+                "DNX_DEFAULT_LIB",
+                "DNX_APPBASE",
+                "DNX_COMPILATION_SERVER_PORT",
```

Comment:
I don't think this is getting removed - it is not set anywhere in the code so assume this is for the user.